src/FSharp.Data.GraphQL.Client/GraphQLClient.fs

```diff
@@ -82,10 +144,77 @@ module GraphQLClient =
                 with postex -> return rethrow [getex; postex]
         }
 
+    /// Executes an introspection schema request to a GraphQL server.
     let sendIntrospectionRequest client serverUrl httpHeaders = 
         sendIntrospectionRequestAsync client serverUrl httpHeaders
         |> Async.RunSynchronously
 
-    let sendRequest client request =
-        sendRequestAsync client request
+    /// Executes a multipart request to a GraphQL server asynchronously.
+    let sendMultipartRequestAsync (connection : GraphQLClientConnection) (request : GraphQLRequest) =
+        async {
+            let client = connection.Client
+            let boundary = sprintf "----GraphQLProviderBoundary%s" (Guid.NewGuid().ToString("N"))
+            use content = new MultipartContent("form-data", boundary)
+            let variables = 
+                request.Variables
+                |> Array.map (fun (name, value) ->
+                    match value with
+                    | null -> name, null
+                    | :? Upload -> name, null
+                    | :? IEnumerable<Upload> as x -> name, x |> Seq.map (fun _ -> null) |> box
+                    | _ -> name, value)
+            let files = 
+                let isUpload (value : obj) =
+                    match value with
+                    | null -> false
+                    | :? Upload | :? IEnumerable<Upload> -> true
+                    | _ -> false
+                let rec mapper (name: string, value : obj) =
+                    match value with
+                    | :? IEnumerable<Upload> as x -> x |> Seq.mapi (fun ix x -> sprintf "%s.%i" name ix, x) |> Array.ofSeq
+                    | _ -> [| name, value :?> Upload |]
+                request.Variables
+                |> Array.filter (fun (_, value) -> isUpload value)
+                |> Array.collect mapper
```

Comment:
These could be merged into a single `Array.collect` like so:
```
|> Array.collect (fun (name, value) -> if isUpload value then mapper (name, value) else [| |])
```

Reply:
I did some changes in the code, but updated them as well based on your suggestion.